src/library/vicon_estimator.cpp

```diff
@@ -442,85 +510,70 @@ void RotationalEstimator::updateEstimateRecombineErrorGlobal(
   *dx_measurement = Eigen::Matrix<double, 6, 1>::Zero();
 }
 
-bool RotationalEstimator::detectMeasurementOutlier(
-    const Eigen::Quaterniond& orientation_measured) {
-  /*
-   *  Method 1 - Detecting the error between the current estimate and the
-   *measurement.
-   *
-   *             Here we compute the rotation between the state estimate and the
-   *current
-   *             measurement. If the magnitude of this rotation is too large we
-   *deem the
-   *             the measurement an outlier and disgard it.
-   *
-   *             The problem with this method is that at initialization the
-   *estimated
-   *             state and measurement can be legitimate different very
-   *different.
-   *             TODO(millanea): The solution to this problem, and probably the
-   *correct way
-   *             to approach this outlier rejection problem, is to compute the
-   *likelihood of
-   *             the measurement given the current estimate of the state and its
-   *variance.
-   */
-  // Constructing the quaternion representing the body to measurement rotation
-  // Eigen::Quaterniond rotation_Z_W = orientation_measured *
-  // orientation_estimate_B_W_.inverse();
-  // Calculating a measure of the size of the error quaternion
-  // double rotation_magnitude_radians =
-  // quaternionRotationMagnitude(rotation_Z_W);
-  /*
-   *  Method 2 - Detecting the error between subsequent measurements.
-   *
-   *             Here we compute the rotation between the current measurement
-   *and the measurement
-   *             from the last time step. If the magnitude of this rotation is
-   *too large we deem the
-   *             the measurement an outlier and disgard it.
-   */
-  // Performing some initialization if this is the first measurement.
-  // Assuming first measurement valid, saving it and returning.
-  if (first_measurement_flag_) {
-    first_measurement_flag_ = false;
+bool RotationalEstimator::checkForEstimatorCrash() {
+  // Testing if covariance matrix member is nan.
+  double nan_test = covariance_(0, 0);
```

Comment:
Yeah I though about doing covariance_.sum() with the assumption that if theres a single NaN in there the sum will also be Nan. I was just wondering if my assumption there is correct. Difficult to test as the estimator no longer crashes ;). I'll put the sum in there and hope it works.